### PR TITLE
DO NOT REVIEW PROGRAMMES-5737 bugfixes

### PR DESCRIPTION
**Pull request by votemike**
Fixing these two problems....

https://jira.dev.bbc.co.uk/browse/PROGRAMMES-5737?focusedCommentId=3610913&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-3610913

Do not merge before https://github.com/bbc/programmes-pages-service/pull/204
I also renamed `lastOnNotAvailableYet()`

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/Controller/FindByPid/TlecController.php [hunk 1]
```diff
@@ -52,10 +52,13 @@ public function __invoke(
             $clips = $aggregationService->findDescendantClips($programme, 4);
         }
 
-        $upcomingEpisodes = null;
+        $hasUpcomingEpisodes = false;
         $streamableEpisodes = null;
         if ($programme->getAggregatedEpisodesCount() > 0) {
             $streamableEpisodes = $aggregationService->findStreamableDescendantEpisodes($programme, 1);
+            if (empty($streamableEpisodes)) {
+                $hasUpcomingEpisodes = null !== $aggregationService->findUpcomingStreamableDescendantEpisodes($programme, 1); //@TODO maybe change to count
```
Comment (votemike) [inline src/Controller/FindByPid/TlecController.php]: Should I change this to be a count method? Or does a count method already exist?

File: src/DsAmen/Organism/Map/SubPresenter/OnDemandPresenter.php [hunk 2]
```diff
@@ -66,7 +80,7 @@ public function getClass(): string
         return $this->class;
     }
 
-    public function getUpcomingEpisode(): ?Episode
+    public function getPendingEpisode(): ?Episode
```
Comment (votemike) [inline src/DsAmen/Organism/Map/SubPresenter/OnDemandPresenter.php]: Happy with this new name?

File: src/Controller/FindByPid/TlecController.php (hunk 1 above)
Comment (felipeparaujo) [reply src/Controller/FindByPid/TlecController.php]: I don't think there's a count for this already. A count would be faster, although probably not by much. You should evaluate it.

File: src/DsAmen/Organism/Map/SubPresenter/OnDemandPresenter.php (hunk 2 above)
Comment (felipeparaujo) [reply src/DsAmen/Organism/Map/SubPresenter/OnDemandPresenter.php]: If you leave a little comment explaining what a pending episode is just above the method, yeah.

Comment (votemike) [reply src/DsAmen/Organism/Map/SubPresenter/OnDemandPresenter.php]: I have added comments here: https://github.com/bbc/programmes-frontend/pull/243/files#diff-d04f71ad0d6c828a552d1ed9031238fbR34 and here: https://github.com/bbc/programmes-frontend/pull/243/files#diff-d04f71ad0d6c828a552d1ed9031238fbR142 . Is that good enough? Or do you want it here too?

File: src/Controller/FindByPid/TlecController.php [hunk 3]
```diff
@@ -52,10 +52,13 @@ public function __invoke(
             $clips = $aggregationService->findDescendantClips($programme, 4);
         }
 
-        $upcomingEpisodes = null;
+        $hasUpcomingEpisodes = false;
         $streamableEpisodes = null;
         if ($programme->getAggregatedEpisodesCount() > 0) {
             $streamableEpisodes = $aggregationService->findStreamableDescendantEpisodes($programme, 1);
+            if (empty($streamableEpisodes)) {
+                $hasUpcomingEpisodes = $aggregationService->countUpcomingStreamableDescendantEpisodes($programme) > 0;
```
Comment (BjornTwachtmann) [inline src/Controller/FindByPid/TlecController.php]: I've looked at the logic for this in V2 and I wonder if this query is necessary. https://github.com/bbc/programmes/blob/develop/webapp/php/lib/BBC/Programmes/View/Map/OnDemand.php#L64 . Could we just look at $upcomingBroadcast->isStreamable() (below)?

Comment (BjornTwachtmann) [reply src/Controller/FindByPid/TlecController.php]: It's probably not a 100% match. But it's probably close enough that nobody will ever notice

Comment (BjornTwachtmann) [reply src/Controller/FindByPid/TlecController.php]: Or rather $upcomingBroadcast->getProgrammeItem()->getStreamableFrom() && !$upcomingBroadcast->getProgrammeItem()->isStreamable()

Comment (votemike) [reply src/Controller/FindByPid/TlecController.php]: Should we OK this with Cathal, Claudio or the testers?

Comment (votemike) [reply src/Controller/FindByPid/TlecController.php]: Thinking of something like this? https://github.com/bbc/programmes-frontend/compare/PROGRAMMES-5737-bugfixes-2?expand=1

**Pull request closed without merging**